[doc] Example of self-reference in scoped packages

piranna opened the pull request:
Original code by @targos from
https://github.com/nodejs/node/issues/37618#issuecomment-791888177

Closes #37618


<!--
Before submitting a pull request, please read
https://github.com/nodejs/node/blob/master/CONTRIBUTING.md.

Commit message formatting guidelines:
https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

targos commented:
Please remove the "@" reference to me otherwise I'll get a notification everytime this commit is pushed somewhere.

piranna commented:
> Please remove the "@" reference to me from the commit message otherwise I'll get a notification everytime this commit is pushed somewhere.

Sorry about that, just only wanted to give you attribution. I'll fix it.

piranna commented:
> Please remove the "@" reference to me from the commit message otherwise I'll get a notification everytime this commit is pushed somewhere.

Fixed :-)

I've also fixed the linting and now all the tests are passing.

aduh95 commented on doc/api/packages.md:
````diff
@@ -683,6 +683,32 @@ and in a CommonJS one. For example, this code will also work:
+console.log(require('@my/package'));
+```
+
+```bash
+node other.js
+# 42
````
````suggestion
```console
> node other.js
42
````

piranna replied:
Can you explain that? I copied the bash code blocks from other places. Can you point me to a console code block?

aduh95 replied:
Sure, here's one: https://github.com/nodejs/node/blob/674614b3f5245f1b5cae974af1365fd632190b41/doc/api/modules.md#L343-L353

piranna replied:
Great, fixed :-)

There are 100 files that are using the `bash` language in code blocks, maybe also related to `npm` or other dependencies. Should we change the Node.js ones in another PR, or is it ok to left them as they are currently?

aduh95 replied:
Here's how we use those in Node.js:
`bash` is a valid language code block as long as it describes a list of instructions or an actual shell script.
`console` is more relevant when you want to show both the command and its output.

We usually don't touch the files from our dependencies, it's OK if they use different conventions than ours; those can stay as is. For Node.js ones, if you see somewhere the wrong language block is being used, of course a PR is welcome :)

piranna replied:
> `bash` is a valid language code block as long as it describes a list of instructions or an actual shell script.
> `console` is more relevant when you want to show both the command and its output.

Great, thanks for the clarification, makes sense :-)


> We usually don't touch the files from our dependencies, it's OK if they use different conventions than ours; those can stay as is.

Obviously, I would not touch the ones for the dependencies, because they would be overwritting when updating them... :-)

targos commented:
@nodejs/modules

piranna commented:
Can this be merged?

aduh95 commented:
Landed in fe4d562dd95fb43961e43ec9b3799660b48e149e.

piranna commented:
Great, thank you! :-D

Trott commented:
Because of some changes in the code formatting that happened in the interim, this is causing some (trivial-to-fix) lint issues. Fix is in https://github.com/nodejs/node/pull/39792. Collaborators, please review/fast-track. Thanks.

Trott commented:
> Because of some changes in the code formatting that happened in the interim, this is causing some (trivial-to-fix) lint issues. Fix is in #39792. Collaborators, please review/fast-track. Thanks.

(I believe the commit-queue would have caught/flagged this, by the way, if that encourages its use more.)
